lib/pg_party/adapter_decorator.rb

```diff
@@ -146,7 +147,23 @@ def quote(value)
     end
 
     def quote_partition_key(key)
-      key.to_s.split("::").map(&method(:quote_column_name)).join("::")
+      if key.is_a?(Proc)
+        key.call.to_s # very difficult to determine how to sanitize a complex expression
+      else
+        quote_column_name(key)
+      end
+    end
+
+    def quote_range_constraint(value)
+      "(#{Array.wrap(value).map(&method(:quote)).join(",")})"
+    end
+
+    def quote_list_constraint(value)
+      if value.is_a?(Array)
+        "(#{value.map(&method(:quote)).join(",")})"
```

Comment:
could `"(#{Array.wrap(value).map(&method(:quote)).join(",")})"` be abstracted into its own method

```ruby
def quote_collection(value)
  "(#{Array.wrap(value).map(&method(:quote)).join(",")})"
end

def quote_range_constraint(value)
  quote_collection(value)
end

def quote_list_constraint(value)
  if value.is_a?(Array)
    quote_collection(value)
  else
    quote(value)
  end
end
```